### PR TITLE
Feature/set up env vars

### PR DESCRIPTION
**Pull request by ConnorDownes**
Not a very interesting PR. Mainly an odd job of tasks and small changes which should have probably been split up into different PRs.

- Installed typescript for use in the project
- Removed custom CSS files for views in favour of scoped style blocks
- Added env vars for use in Vue, will allow us to deploy staging/production instances in the future
- Implemented some 'services' which contain axios requests (Deck, Card)
- Fixed some console errors so it should be a clean sheet now
- Styling changes
- Extracted some core functionality which can be re-used into components (Error box, loader)

### CODE REVIEW

File: src/MTGCore.ClientApp/ClientApp/src/views/Decks.vue [hunk 1]
```diff
@@ -1,170 +1,157 @@
...
+import CardList from "../components/CardList.vue";
+import DeckList from "../components/DeckList.vue";
+import Loader from "../components/base/Loader.vue";
+import ErrorMessage from "../components/base/ErrorMessage.vue";
+import DeckService from "../services/deck";
+import axios from 'axios';
+import Vue from "vue";
 
-import '../assets/css/Decks.css';
+interface Data {
```
Comment (StephenGibbsDev) [inline src/MTGCore.ClientApp/ClientApp/src/views/Decks.vue:65]: move into its own file.

or create class instead and use that for the data

**Review by StephenGibbsDev — CHANGES_REQUESTED** [review]
LGTM except for a few concerns / recommendations

File: src/MTGCore.ClientApp/ClientApp/src/main.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import router from './router/index'
+import CardService from "./services/card";
+import DeckService from "./services/deck";
+
+Vue.config.productionTip = false
+
+const baseUrl = process.env.VUE_APP_API_BASE_URL;
+if (!baseUrl) throw new Error('Base url is not set-up');
+    
+CardService.init(baseUrl);
```
[the remark this answers is not in the document]
Comment (ConnorDownes) [reply src/MTGCore.ClientApp/ClientApp/src/main.ts:12]: Yeah. Vuex will probably be handy for storing the user login state which I plan on implementing in the next couple of days too.

**Review by StephenGibbsDev — APPROVED** [review]
LGTM

**Pull request merged**